lib/graphite.py

```diff
@@ -0,0 +1,33 @@
+import socket
+import time
+
+
+class Graphite(object):
+
+    def __init__(self, hostname, port):
+        self.hostname = hostname
+        self.port = int(port)
+
+    def flatten_dict(self, d):
+        def expand(key, value):
+            if isinstance(value, dict):
+                return [(key + '.' + k, v) for k, v in self.flatten_dict(value).items()]
+            else:
+                return [(key, value)]
+        items = [item for k, v in d.items() for item in expand(k, v)]
+        return dict(items)
+
+    def update(self, prefix, metrics, nodes):
+        ts = int(time.time())
+
+        sock = socket.socket()
+        sock.connect((self.hostname, self.port))
+
+        for node in nodes:
+            if node['flags']['online'] and node['nodeinfo']['node_id']:
+                stats = self.flatten_dict(node['statistics'])
+                for metric in metrics.split(','):
+                    if stats[metric]:
+                        msg = prefix + '%s.%s %s %d\n' % (node['nodeinfo']['node_id'], metric.replace(' ', '_'), stats[metric], ts)
+                        sock.send(msg.encode())
```

Comment:
I know this defaults to 'utf-8' because I looked it up. Shouldn't we be explicit here?